ocaml/typing/mode_intf.mli

```diff
@@ -465,28 +467,49 @@ module type S = sig
     module Value : sig
       type atom := t
 
+      type error =
+        | Error : ('m, 'a, _) Value.axis * ('m, 'a) raw Solver.error -> error
+
+      type nonrec equate_error = equate_step * error
+
+      module Const : sig
+        (** A modality that acts on [Value] modes. Conceptually it is a sequnce
+            of [atom] that acts on individual axes. *)
+        type t
+
+        (** The identity modality. *)
+        val id : t
+
+        (** Apply a modality on mode. *)
+        val apply : t -> ('l * 'r) Value.t -> ('l * 'r) Value.t
+
+        (** [compose m t] returns the modality that is [m] after [t]. *)
+        val compose : atom -> t -> t
```

Comment:
Might be worth naming one or both of the arguments so that it's clear which one applies first (I'm partial to calling the second one `then_` and leaving the first one anonymous).